jvb/src/main/kotlin/org/jitsi/videobridge/octo/config/OctoConfig.kt

```diff
@@ -16,151 +16,75 @@
+
+    // We grab these two properties from the legacy config separately here
+    // because we use them to infer a legacy value of 'enabled' (which was
+    // based on the presence of these properties) and as potential values
+    // in each of the individual bindAddress and bindPort properties.
+    private val legacyBindAddress: String? by optionalconfig(
+        "org.jitsi.videobridge.octo.BIND_ADDRESS".from(JitsiConfig.legacyConfig))
+    private val legacyBindPort: Int? by optionalconfig(
+        "org.jitsi.videobridge.octo.BIND_PORT".from(JitsiConfig.legacyConfig))
+
+    val enabled: Boolean by config {
+        // The legacy config file doesn't have an 'enabled' property,
+        // instead it was based on the values of the parameters.  Here,
+        // we simulate a legacy 'enabled' value based on the results
+        // of validating the other properties in the legacy config
+        // file.  If neither property is present, we consider the field
+        // "not found" in the legacy config.
+        "Legacy Octo relay enabled" {
+            if (legacyBindAddress == null && legacyBindPort == null) {
+                throw ConfigException.UnableToRetrieve.NotFound("not found")
             }
+            legacyBindAddress != null && legacyBindPort?.isUnprivilegedPort() == true
+        }
+        "videobridge.octo.enabled".from(JitsiConfig.newConfig)
+    }
 
-            class BindAddressProperty : LegacyFallbackConfigProperty<String>(
-                String::class,
-                "org.jitsi.videobridge.octo.BIND_ADDRESS",
-                "videobridge.octo.bind-address",
-                readOnce = true
-            )
-
-            private val bindAddressProp = BindAddressProperty()
-
-            @JvmStatic
-            fun bindAddress(): String = bindAddressProp.value
-
-            class BindPortProperty : FallbackProperty<Int>(
-                legacyConfigAttributes {
-                    name("org.jitsi.videobridge.octo.BIND_PORT")
-                    readOnce()
-                    retrievedAs<Int>() convertedBy {
-                        if (!it.isUnprivilegedPort()) {
-                            throw ConfigValueParsingException("Octo bind port " +
-                                    "must be in the unprivileged port space")
-                        }
-                        it
-                    }
-                },
-                newConfigAttributes {
-                    name("videobridge.octo.bind-port")
-                    readOnce()
-                }
-            )
-
-            private val bindPortProp = BindPortProperty()
+    val region: String? by optionalconfig {
+        "org.jitsi.videobridge.REGION".from(JitsiConfig.legacyConfig)
+        "videobridge.octo.region".from(JitsiConfig.newConfig)
+    }
 
-            @JvmStatic
-            fun bindPort(): Int = bindPortProp.value
+    val bindAddress: String by config {
+        "bind address from legacy config" { legacyBindAddress!! }
```

Comment:
And here, since we use `by config` rather than `by optionalconfig`, if `legacyBindAddress == null` the first retrieval will fail, so we fallback to the new property?

Reply:
Right